### PR TITLE
Add Description to describe cmd to show description in annotations

### PR DESCRIPTION
**Pull request by zherman0**
**What this PR does / why we need it**:
Add Description to describe cmd to show description in annotations
**Which issue this PR fixes** _(optional, in `fixes #<issue number>(, #<issue_number>, ...)` format, will close that issue when PR gets merged)_: fixes #

**Special notes for your reviewer**:
@eparis @saad-ali 
**Release note**:

``` release-note
```

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/35295)

### CODE REVIEW

**Comment by eparis** [comment]
@jsafrane

**Comment by eparis** [comment]
This is a bit strange. Do we show 'annotations' in describe for any other object? Now we are going to show the description twice. Once in the annotations field and once in a nice way...

Should we get rid of the annotations in describe?

**Comment by eparis** [comment]
@screeley44

**Comment by eparis** [comment]
@k8s-bot test this

**Comment by eparis** [comment]
This is the first place we are trying to materialize the 'description' into `kubectl describe`. Looking for feedback on if we should display annotations at all in kubectl describe (I think no, don't think other objects do this) and I think yes to the concept...

@kubernetes/sig-cli

**Comment by screeley44** [comment]
@eparis - in 34638 that merged upstream, I did still show both the annotations and pretty (for is-default-class ann) IsDefaultClass label.  

I think for storage-class describer we  could make an argument to not show annotations at all, since we represent them anyway in a pretty label and if users want to see real annotations, they can always -o yaml to see the real stuff

File: pkg/apis/storage/util/helpers.go [hunk 1]
```diff
@@ -45,6 +45,8 @@ const BetaStorageClassAnnotation = "volume.beta.kubernetes.io/storage-class"
 //   volume belongs.
 //TODO: Update this to final annotation value as it matches BetaStorageClassAnnotation for now
 const StorageClassAnnotation = "volume.beta.kubernetes.io/storage-class"
+const StorageClassKubernetesDescription = "kubernetes.io/description"
+const StorageClassDescription = "description"
```
Comment (eparis) [inline pkg/apis/storage/util/helpers.go]: I wonder if these definitions and the helpers belong in a more generic place, for future use in other object types...

**Comment by screeley44** [comment]
@eparis - and furthermore, the way annotations are displayed are kinda ugly, especially if you have more than one, for example, set the default and a description and if description is long

```
IsDefaultClass:	Yes
Annotations:	kubernetes.io/descritpion=this is a test,storageclass.beta.kubernetes.io/is-default-class=true
```

**Comment by eparis** [comment]
@zherman0 lets do that, move the const defintions and helper somewhere more generic and drop showing the annotations list altogether...

**Comment by zherman0** [comment]
We are going to redo this as more generic in another PR, so closing this one

**Pull request closed without merging**